### PR TITLE
Add tests for parsing a hash-name reference

### PR DESCRIPTION
**Pull request by zcorpan**
See https://github.com/whatwg/html/issues/2239#issuecomment-270605836

### CODE REVIEW

**Comment by zcorpan** [comment]
Many failures in Gecko are due to `elementFromPoint` not returning `area` elements.

https://bugzilla.mozilla.org/show_bug.cgi?id=1227469

**Comment by zcorpan** [comment]
<details>
<summary>Edge 13: 24 Pass, 24 Fail</summary>
win10_edge_13 0
</details>

<details>
<summary>IE 11: 26 Pass, 22 Fail</summary>
win10_ie_11 0
</details>

**Comment by zcorpan** [comment]
It appears all browsers at some point start returning `null` from `elementFromPoint`. Unclear to me what's up with that.

**Comment by zcorpan** [comment]
`null` issue was because the images were outside the viewport.

**Comment by zcorpan** [comment]
https://dxr.mozilla.org/mozilla-central/rev/7011ed1427de2b6f075c46cc6f4618d3e9fcd2a4/dom/base/nsDocument.cpp#9949

Gecko's implementation is exactly like the spec, as far as I can tell... except the spec doesn't say what to return if there is no `map` element with the given name/id. Will PR to fix the spec for that.

Anyway, this PR is ready for review.

File: html/semantics/embedded-content/image-maps/image-map-processing-model/hash-name-reference-test-data.html [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+ </map>
+</div>
+
+<div data-expect="area-two-maps-with-this-id-1">
+ <img src="/images/threecolors.png" usemap="#two-maps-with-this-id"/>
+ <object data="/images/threecolors.png" usemap="#two-maps-with-this-id"></object>
+ <map id="two-maps-with-this-id">
+  <area shape="rect" coords="0,0,99,50" href="#area-two-maps-with-this-id-1"/>
+ </map>
+ <map id="two-maps-with-this-name">
```
Comment (domenic) [inline html/semantics/embedded-content/image-maps/image-map-processing-model/hash-name-reference-test-data.html]: Should this be two-maps-with-this-id?

**Review by domenic — APPROVED** [review]
LGTM with one mistake, I think...

**Pull request merged**